integration_tests/src/main/python/cache_test.py

```diff
@@ -61,9 +61,6 @@ def test_passing_gpuExpr_as_Expr():
 @pytest.mark.parametrize('join_type', ['Left', 'Right', 'Inner', 'LeftSemi', 'LeftAnti'], ids=idfn)
 @ignore_order
 def test_cache_join(data_gen, join_type):
-    if data_gen.data_type == BooleanType():
```

Comment:
Isn't this still an XFail for spark 3.0.0?  The fix went into spark 3.0.1

Reply:
hmm... I tested with the 3.0.0 branch as well. But I get what you are saying, the release doesn't have the fix. 

I can add a method in the spark_session.py to support version based checks. Is that acceptable?

Reply:
yes